flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/aggregate/RowTimeBoundedRowsOver.java

```diff
@@ -0,0 +1,292 @@
+import org.apache.flink.api.common.state.MapStateDescriptor;
+import org.apache.flink.api.common.state.ValueState;
+import org.apache.flink.api.common.state.ValueStateDescriptor;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.typeutils.ListTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction;
+import org.apache.flink.table.api.TableConfig;
+import org.apache.flink.table.dataformat.BaseRow;
+import org.apache.flink.table.dataformat.JoinedRow;
+import org.apache.flink.table.dataview.PerKeyStateDataViewStore;
+import org.apache.flink.table.generated.AggsHandleFunction;
+import org.apache.flink.table.generated.GeneratedAggsHandleFunction;
+import org.apache.flink.table.type.InternalType;
+import org.apache.flink.table.typeutils.BaseRowTypeInfo;
+import org.apache.flink.util.Collector;
+import org.apache.flink.util.Preconditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Process Function for ROWS clause event-time bounded OVER window.
+ */
+public class RowTimeBoundedRowsOver<K> extends
+	ProcessFunctionWithCleanupState<K, BaseRow, BaseRow> {
+	private static final Logger LOG = LoggerFactory.getLogger(RowTimeBoundedRowsOver.class);
+
+	private GeneratedAggsHandleFunction genAggsHandler;
+	private InternalType[] accTypes;
+	private InternalType[] inputFieldTypes;
+	private Long precedingOffset;
+	private int rowTimeIdx;
+
+	private JoinedRow output;
+
+	// the state which keeps the last triggering timestamp
+	private ValueState<Long> lastTriggeringTsState;
+
+	// the state which keeps the count of data
+	private ValueState<Long> counterState;
+
+	// the state which used to materialize the accumulator for incremental calculation
+	private ValueState<BaseRow> accState;
+
+	// the state which keeps all the data that are not expired.
+	// The first element (as the mapState key) of the tuple is the time stamp. Per each time stamp,
+	// the second element of tuple is a list that contains the entire data of all the rows belonging
+	// to this time stamp.
+	private MapState<Long, List<BaseRow>> inputState;
+
+	private AggsHandleFunction function;
+
+	public RowTimeBoundedRowsOver(
+		GeneratedAggsHandleFunction genAggsHandler,
+		InternalType[] accTypes,
+		InternalType[] inputFieldTypes,
+		Long precedingOffset,
+		int rowTimeIdx,
+		TableConfig tableConfig) {
+		super(tableConfig);
+		Preconditions.checkNotNull(precedingOffset);
+		this.genAggsHandler = genAggsHandler;
+		this.accTypes = accTypes;
+		this.inputFieldTypes = inputFieldTypes;
+		this.precedingOffset = precedingOffset;
+		this.rowTimeIdx = rowTimeIdx;
+	}
+
+	@Override
+	public void open(Configuration parameters) throws Exception {
+		LOG.debug("Compiling AggregateHelper: " + genAggsHandler.getClassName() + " \n\n" +
```

Comment:
Actually, we don't need this anymore, because `genAggsHandler.newInstance` will log the code.